integrations/uniswap/src/evm-contract-events.ts

```diff
@@ -0,0 +1,98 @@
+import { DataItem, IRuntime, Validator, sha256 } from '@kyvejs/protocol';
+import { providers, utils } from 'ethers';
+
+// Method to get the named args
+const parseArgs = (struct: any) => {
+  const parsedStruct = { ...struct };
+  const keyLength = Object.keys(struct).length / 2;
+
+  for (let i = 0; i < keyLength; i++) {
+    delete parsedStruct[i];
+  }
+
+  return parsedStruct;
+};
+
+import { name, version } from '../package.json';
+
+export default class Evm implements IRuntime {
+  public name = name;
+  public version = version;
+
+  async getDataItem(
+    v: Validator,
+    source: string,
+    key: string
+  ): Promise<DataItem> {
+    const apiObj = JSON.parse(process.env.API_KEYS!);
+    // setup web3 provider
+    const provider = new providers.StaticJsonRpcProvider(
+      source + apiObj[source]
+    );
+
+    // try to fetch data item
+    const value = await provider.getLogs({
+      address: v.poolConfig.contract.address,
+      fromBlock: parseInt(key),
+      toBlock: parseInt(key),
+    });
+
+    // throw if data item is not available
+    if (!value) throw new Error();
```

Comment:
this can be removed since `prevalidateDataItem` already checks if value is defined